styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java

```diff
@@ -92,78 +87,68 @@ public WorkflowResource(Storage storage, String schedulerServiceBaseUrl,
             rc -> patchState(arg("cid", rc), arg("wfid", rc), rc.request())),
         Route.with(
             json(), "PATCH", BASE + "/<cid>/state",
-            rc -> patchState(arg("cid", rc), rc.request())),
-        Route.with(
-            json(), "POST", BASE + "/<cid>",
+            rc -> patchState(arg("cid", rc), rc.request()))
+
+    );
+
+    final List<Route<AsyncHandler<Response<ByteString>>>> forwardedRoutes = Arrays.asList(
+        Route.async(
+            "POST", BASE + "/<cid>",
             rc -> createOrUpdateWorkflow(arg("cid", rc), rc)
         ),
-        Route.with(
-            json(), "DELETE", BASE + "/<cid>/<wfid>",
+        Route.async(
+            "DELETE", BASE + "/<cid>/<wfid>",
             rc -> deleteWorkflow(arg("cid", rc), arg("wfid", rc), rc)
         )
     );
 
-    return Api.prefixRoutes(routes, V2, V3);
+    return cat(
+        Api.prefixRoutes(routes, V2, V3),
+        Api.prefixRoutes(forwardedRoutes, V3)
+    );
   }
 
-  private Response<ByteString> deleteWorkflow(String cid, String wfid, RequestContext rc) {
+  private CompletionStage<Response<ByteString>> deleteWorkflow(String cid, String wfid,
+                                                               RequestContext rc) {
     Request request = Request.forUri(schedulerApiUrl("workflows", cid, wfid), "DELETE");
-    try {
-      return rc.requestScopedClient().send(request).toCompletableFuture().get();
-    } catch (InterruptedException | ExecutionException e) {
-      return Response
-          .forStatus(Status.INTERNAL_SERVER_ERROR.withReasonPhrase("Error in call to Scheduler"));
-    }
+    return rc.requestScopedClient().send(request);
```

Comment:
Something like `rc.requestScopedClient().send(rc.request().withUri(schedulerApiUrl("workflows", cid, wfid)));` instead to keep additional header, etc.